sdk_contrib/spring_boot2/src/main/java/io/opentelemetry/contrib/spring/boot/TracerSdkProviderBean.java

```diff
@@ -0,0 +1,266 @@
+  private Clock otelClock;
+  private IdsGenerator otelIdsGenerator;
+  private Resource otelResource;
+  private TracerProvider tracerProvider;
+  private final List<SpanProcessor> spanProcessors = new ArrayList<>();
+  private final List<SpanExporter> spanExporters = new ArrayList<>();
+
+  /**
+   * Sets the Spring application properties used to configure the OpenTelemetry SDK.
+   *
+   * @param properties the properties
+   */
+  @Autowired
+  public void setProperties(OpenTelemetryProperties properties) {
+    this.properties = properties;
+  }
+
+  /**
+   * Sets the Spring-managed OpenTelemetry tracing {@link Sampler} implementation.
+   *
+   * @param sampler the sampler
+   */
+  @Autowired(required = false)
+  public void setSampler(Sampler sampler) {
+    this.sampler = sampler;
+  }
+
+  /**
+   * Sets the Spring-managed OpenTelemetry clock implementation for generating span timestamps.
+   *
+   * @param otelClock the clock
+   */
+  @Autowired
+  public void setOtelClock(Clock otelClock) {
+    this.otelClock = otelClock;
+  }
+
+  /**
+   * Sets the Spring-managed OpenTelemetry tracing IDs generator.
+   *
+   * @param otelIdsGenerator the generator
+   */
+  @Autowired
+  public void setOtelIdsGenerator(IdsGenerator otelIdsGenerator) {
+    this.otelIdsGenerator = otelIdsGenerator;
+  }
+
+  /**
+   * Sets the Spring-managed implementation of OpenTelemetry {@link Resource} which populates global
+   * application attributes.
+   *
+   * @param otelResource the resource populator
+   */
+  @Autowired
+  @Qualifier("otelResource")
+  public void setOtelResource(Resource otelResource) {
+    this.otelResource = otelResource;
+  }
+
+  /**
+   * Sets all Spring-managed OpenTelemetry tracing span processors.
+   *
+   * @param spanProcessors the processors
+   */
+  @Autowired(required = false)
+  public void setSpanProcessors(List<SpanProcessor> spanProcessors) {
+    if (spanProcessors != null) {
+      this.spanProcessors.clear();
+      this.spanProcessors.addAll(spanProcessors);
+    }
+  }
+
+  /**
+   * Sets all Spring-managed OpenTelemetry tracing span exporters.
+   *
+   * @param spanExporters the exporters
+   */
+  @Autowired(required = false)
+  public void setSpanExporters(List<SpanExporter> spanExporters) {
+    if (spanExporters != null) {
+      this.spanExporters.clear();
+      this.spanExporters.addAll(spanExporters);
+    }
+  }
+
+  @Override
+  public void afterPropertiesSet() {
+    tracerProvider = initializeTracerRegistry();
+  }
+
+  @Override
+  public TracerProvider getObject() {
+    if (tracerProvider == null) {
+      tracerProvider = initializeTracerRegistry();
+    }
+    return tracerProvider;
+  }
+
+  @Override
+  public Class<?> getObjectType() {
+    return TracerProvider.class;
+  }
+
+  private TracerProvider initializeTracerRegistry() {
+    TracerSdkProvider registry =
+        TracerSdkProvider.builder()
+            .setClock(otelClock)
+            .setIdsGenerator(otelIdsGenerator)
```

Comment:
Shouldn't the tracer use only [W3C TraceContext](https://www.w3.org/TR/trace-context/) compatible ids? Or is it allow to have different id providers for contrib packages?

Reply:
I need to use AWS X-Ray compatible trace ids. All X-Ray trace ids are compatible with W3C TraceContext but not all TraceContext trace ids are compatible with X-Ray. See: https://github.com/kbrockhoff/opentelemetry-java/blob/master/sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/AwsXRayIdsGenerator.java

Also different ids generator implementation is needed for backward Android compatibility than should be used in server-side JVM applications.